sonarts-core/src/tools/newRule.ts

```diff
@@ -1,269 +1,288 @@
-    replaceDict["</h2>"] = `\n`;
-
-    const fileText = fs.readFileSync(`${rspecRuleFolder}/${rspecId}.html`).toString();
-    return replace(fileText, replaceDict);
-  } catch (err) {
-    console.error("could not find metadata for the rule. Have you run the rspec-api?");
-    showHelp();
-    throw err;
-  }
-}
+      console.log("ruleLinks");
+      console.log(ruleLinks);
 
-function updateReadme(rootFolder: string, ruleTitle: string, ruleNameDash: string) {
-  const readmePath = `${rootFolder}/README.md`;
-  let readme = fs.readFileSync(readmePath).toString();
-
-  const lines = readme.split("\n");
-
-  let result = [];
-
-  let resultsBlock1 = [];
-  let resultsBlock2 = [];
-  let resultsBlock3 = [];
-
-  let ruleValue = `* ${ruleTitle} ([\`${ruleNameDash}\`])`;
-  resultsBlock1.push(ruleValue);
-  let ruleKey = `[\`${ruleNameDash}\`]: ./sonarts-core/docs/rules/${ruleNameDash}.md`;
-  resultsBlock2.push(ruleKey);
-
-  // 0: start
-  // 1: ## Rules
-  // 2: newlines
-  // 3: rules
-  // 4: newlines
-  // 5: rules 2
-  // 6: newline
-  // 7: after
-
-  let state = 0;
-
-  for (const line of lines) {
-    switch (state) {
-      case 0:
-        if (line.trim() === "## Rules") {
-          state = 1;
-        }
-        result.push(line);
-        break;
-
-      case 1:
-      case 2:
-        if (line.length > 0) {
-          state++;
-          resultsBlock1.push(line);
-        } else {
-          result.push(line);
-        }
-
-        break;
-
-      case 3:
-        if (line.length === 0) {
-          state++;
-        } else {
-          resultsBlock1.push(line);
-        }
-        break;
-
-      case 4:
-        if (line.length === 0) {
-          state++;
-        } else {
-          resultsBlock2.push(line);
-        }
-        break;
-      case 5:
-        resultsBlock3.push(line);
-        break;
+      throw new Error("Could not parse README.md.");
     }
-  }
 
-  if (resultsBlock1.length !== resultsBlock2.length) {
-    console.log("resultsBlock1");
-    console.log(resultsBlock1);
+    ruleTitles.push(`* ${ruleTitle} ([\`${ruleNameDash}\`])`);
+    ruleLinks.push(`[\`${ruleNameDash}\`]: ./sonarts-core/docs/rules/${ruleNameDash}.md`);
 
-    console.log("resultsBlock2");
-    console.log(resultsBlock2);
+    const linksToTitles: { [x: string]: string } = {};
+    for (let i = 0; i < ruleTitles.length; i++) {
+      linksToTitles[ruleLinks[i]] = ruleTitles[i];
+    }
 
-    throw new Error("could not parse README.md!");
-  }
+    const sortedRuleLinks = Object.keys(linksToTitles).sort();
+    const sortedRuleTitles = sortedRuleLinks.map(ruleKey => linksToTitles[ruleKey]);
 
-  let dict2: any = {};
-  for (let i = 0; i < resultsBlock1.length; i++) {
-    dict2[resultsBlock2[i]] = resultsBlock1[i];
+    fs.writeFileSync(readmePath, [...head, ...sortedRuleTitles, "", ...sortedRuleLinks, "", ...tail].join("\n"));
   }
 
-  let keys = Object.keys(dict2).sort();
-  let values = new Array();
+  function parseReadme() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Refactor this function to reduce its Cognitive Complexity from 18 to the 15 allowed. ([typescript:S3776](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3776))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&branch=stas%2Frefactor-new-rule&issues=AWDG5FhuJC-F6PRGRIFW&open=AWDG5FhuJC-F6PRGRIFW)

[ISSUE_KEY]: AWDG5FhuJC-F6PRGRIFW